backend/mobilityAI/Controllers/DevicesController.cs

```diff
@@ -35,10 +35,14 @@ public class DevicesController : ControllerBase
         [HttpGet("{deviceId}")]
         public JsonResult GetDevice(string deviceId)
         {
-            var data = from a in _context.Devices
-                       join b in _context.Users on a.PatientID equals b.Id
-                       where (a.Id == deviceId)
-                       select new { a.FriendlyName, b.FirstName, b.LastName, a.LastSync };
+            var data = (from a in _context.Devices
+                        join b in _context.Users on a.PatientID equals b.Id
+                        where (a.Id == deviceId)
+                        select new { a.Id, a.FriendlyName, b.FirstName, b.LastName, a.LastSync }).SingleOrDefault();
+
+            //Return -1 to indicate that a device was not found
+            if(data == null)
+                return new JsonResult(new Device() { Id = "-1", FriendlyName="", PatientID=-1, LastSync=DateTime.Today });
```

Comment:
Create an issue to have an `Error` return type from the API. Instead of returning -1.

Reply:
Another option would be to return a 500 status code with an explanation.